src/util.rs

```diff
@@ -12,32 +12,26 @@ use core::array::IntoIter;
 use generic_array::typenum::U0;
 use generic_array::{ArrayLength, GenericArray};
 
-use crate::errors::InternalError;
+use crate::{Error, Result};
 
 // Corresponds to the I2OSP() function from RFC8017
-pub(crate) fn i2osp<L: ArrayLength<u8>>(
-    input: usize,
-) -> Result<GenericArray<u8, L>, InternalError> {
+pub(crate) fn i2osp<L: ArrayLength<u8>>(input: usize) -> Result<GenericArray<u8, L>> {
     const SIZEOF_USIZE: usize = core::mem::size_of::<usize>();
 
-    // Check if input >= 256^length
+    // Make sure input fits in output.
     if (SIZEOF_USIZE as u32 - input.leading_zeros() / 8) > L::U32 {
-        return Err(InternalError::SerializationError);
-    }
-
-    if L::USIZE <= SIZEOF_USIZE {
-        return Ok(GenericArray::clone_from_slice(
-            &input.to_be_bytes()[SIZEOF_USIZE - L::USIZE..],
-        ));
+        return Err(Error::SerializationError);
     }
 
     let mut output = GenericArray::default();
-    output[L::USIZE - SIZEOF_USIZE..].copy_from_slice(&input.to_be_bytes());
+    output[L::USIZE.saturating_sub(SIZEOF_USIZE)..]
+        .copy_from_slice(&input.to_be_bytes()[SIZEOF_USIZE.saturating_sub(L::USIZE)..]);
```

Comment:
This is correct as far as I can tell, but I'm pointing it out here to be extra-careful.